File: csharp/src/Movie.cs
namespace RefactoringKata
{
    public class Movie
    {
    public const int CHILDRENS = 2;
    public const int REGULAR = 0;
    public const int NEW_RELEASE = 1;

        private string title;
        private int priceCode;

        public Movie(string title, int priceCode)
        {
            this.title = title;
            this.priceCode = priceCode;
        }

        public int GetPriceCode()
        {
            return priceCode;
        }

        public void SetPriceCode(int priceCode)
        {
            this.priceCode = priceCode;
        }

        public string GetTitle()
        {
            return title;
        }
    }
}

File: typescript/src/movie.ts
const childrens = 2,
  regular = 0,
  newRelease = 1;

class Movie {
  title: String;
  priceCode: number;

  constructor(title: String, priceCode: number) {
    this.title = title;
    this.priceCode = priceCode;
  }

  getPriceCode(): number {
    return this.priceCode;
  }

  setPriceCode(priceCode: number) {
    this.priceCode = priceCode;
  }

  getTitle(): String {
    return this.title;
  }

  static CHILDRENS(): number {
    return childrens;
  }

  static REGULAR(): number {
    return regular;
  }

  static NEW_RELEASE(): number {
    return newRelease;
  }
}

export default Movie;